lca-cli/installationiso/installationiso.go

```diff
@@ -0,0 +1,184 @@
+package installationiso
+
+import (
+	"bytes"
+	"embed"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path"
+	"text/template"
+
+	"github.com/openshift-kni/lifecycle-agent/lca-cli/ops"
+	"github.com/sirupsen/logrus"
+)
+
+type InstallationIso struct {
+	log     *logrus.Logger
+	ops     ops.Ops
+	workDir string
+}
+
+type IgnitionData struct {
+	SeedImage        string
+	SeedVersion      string
+	BackupSecret     string
+	PullSecret       string
+	SshPublicKey     string
+	LCAImage         string
+	InstallationDisk string
+}
+
+func NewInstallationIso(log *logrus.Logger, ops ops.Ops, workDir string) *InstallationIso {
+	return &InstallationIso{
+		log:     log,
+		ops:     ops,
+		workDir: workDir,
+	}
+}
+
+const (
+	ibiButaneTemplateFilePath = "data/ibi-ignition.template"
+	ibiIgnitionFileName       = "ibi-ignition.json"
+	rhcosIsoFileName          = "rhcos-live.x86_64.iso"
+	ibiIsoFileName            = "rhcos-ibi.iso"
+	coreosInstallerImage      = "quay.io/coreos/coreos-installer:latest"
+)
+
+//go:embed data/*
+var f embed.FS
+
+func (r *InstallationIso) Create(seedImage, seedVersion, authFile, pullSecretFile, sshPublicKeyPath, lcaImage, rhcosLiveIsoUrl, installationDisk string) error {
+	r.log.Info("Creating IBI installation ISO")
+	err := r.validate()
+	if err != nil {
+		return err
+	}
+	err = r.renderIgnition(seedImage, seedVersion, authFile, pullSecretFile, sshPublicKeyPath, lcaImage, installationDisk)
+	if err != nil {
+		return err
+	}
```

Comment:
just a nit to your consideration... if so, mind the `err :=` for below conditionals
```suggestion
	if err := r.validate(); err != nil {
		return err
	}
	if err = r.renderIgnition(seedImage, seedVersion, authFile, pullSecretFile, sshPublicKeyPath, lcaImage, installationDisk); err != nil {
		return err
	}
```

Reply:
I think in this case the line gets too long, making it less readable